gradio/processing_utils.py

```diff
@@ -376,9 +376,12 @@ def get_temp_url_path(self, url: str) -> str:
         file_hash = self.hash_url(url)
         return prefix + file_hash + extension
 
-    def get_temp_base64_path(self, base64_encoding: str, prefix: str) -> str:
-        extension = get_extension(base64_encoding)
-        extension = "." + extension if extension else ""
+    def get_temp_base64_path(
+        self, base64_encoding: str, prefix: str, extension: str
+    ) -> str:
+        guess_extension = get_extension(base64_encoding)
+        if not extension and guess_extension:
```

Comment:
The Model3D test was failing because the extension was not being correctly guessed from the base64. This fixes that. I have no idea why the test was passing on CI